src/api/api.js

```diff
@@ -105,70 +102,64 @@ export const constructTags = (tagFilters) => {
   );
 };
 
+const buildOperatingSystemFilter = (osFilterState = {}) => {
+  return Object.entries(
+    Object.values(osFilterState).reduce(
+      (acc, item) => ({ ...acc, ...item }),
+      {}
+    )
+  )
+    .filter(([, value]) => value === true)
+    .map(([key]) => {
+      const keyParts = key.split('-');
+      return [
+        keyParts.slice(0, keyParts.length - 2).join(' '),
+        keyParts[keyParts.length - 1],
+      ];
+    })
+    .reduce((oses, [osName, version]) => {
+      if (!oses[osName]) {
+        oses[osName] = {
+          version: { eq: [] },
+        };
+      }
+      return {
+        ...oses,
+        [osName]: {
+          version: {
+            eq: [...oses[osName].version.eq, version],
+          },
+        },
+      };
+    }, {});
+};
+
 export const calculateSystemProfile = ({
   osFilter,
   rhcdFilter,
   updateMethodFilter,
   hostTypeFilter,
 }) => {
-  let systemProfile = {};
-  const osFilterValues = Array.isArray(osFilter)
-    ? osFilter
-    : Object.values(osFilter || {}).flatMap((majorOsVersion) =>
-        Object.keys(majorOsVersion)
-      );
-
-  if (osFilterValues?.length > 0) {
-    let centosVersions = [];
-    let rhelVersions = [];
-    systemProfile.operating_system = {};
-
-    osFilterValues.forEach((filterValue) => {
-      if (filterValue.osName === 'RHEL') {
-        rhelVersions.push(filterValue.value);
-      }
+  const system_profile = {
+    ...(hostTypeFilter ? { host_type: hostTypeFilter } : {}),
```

Comment:
It doesn't seem to be an equal change. Before, if hostTypeFilter was undefined, we still provided 'nil' value. Won't your change lead to any errors? I am not sure what's the right method here, but just want to point out to this difference